lib/matplotlib/axes/_base.py

```diff
@@ -2103,22 +2104,31 @@ def autoscale(self, enable=True, axis='both', tight=None):
             The *tight* setting is retained for future autoscaling
             until it is explicitly changed.
 
+        This returns a context-manager/decorator that allows temporarily
+        setting the autoscale status, i.e.::
 
-        Returns None.
+            with axes.autoscale(enable): ...
+
+        will keep the autoscale status to `enable` for the duration of the
+        block only.
         """
-        if enable is None:
-            scalex = True
-            scaley = True
-        else:
-            scalex = False
-            scaley = False
+        orig_autoscale = self._autoscaleXon, self._autoscaleYon
+        if enable is not None:
             if axis in ['x', 'both']:
                 self._autoscaleXon = bool(enable)
-                scalex = self._autoscaleXon
             if axis in ['y', 'both']:
                 self._autoscaleYon = bool(enable)
-                scaley = self._autoscaleYon
-        self.autoscale_view(tight=tight, scalex=scalex, scaley=scaley)
+        self.autoscale_view(
```

Comment:
I was just reading this code locally and sorting through what inputs are going to the `autoscale_view`, you are correct this does not actually change the behaviour.  Sorry for the noise.